jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -38,19 +41,259 @@
  */
 
 /**
+ * Helper function to try to parse a part of a date string
+ *
+ * @return NaN if cannot read from string, ToNumber() otherwise
+ */
+static ecma_number_t
+ecma_date_parse_date_chars (lit_utf8_iterator_t *iter, /**< iterator of the utf8 string */
+                            uint32_t num_of_chars) /**< number of characters to read and convert */
+{
+  ecma_number_t ret_value = ecma_number_make_nan ();
+  lit_utf8_size_t copy_size = 0;
+  const lit_utf8_byte_t *str_start_p = iter->buf_p + iter->buf_pos.offset;
+
+  while (num_of_chars--)
+  {
+    if (lit_utf8_iterator_is_eos (iter)
+        || !lit_char_is_unicode_digit (lit_utf8_iterator_peek_next (iter)))
+    {
+      copy_size = 0;
```

Comment:
For me, early return would be better here.

I think that we should avoid early returns in functions, where they reduce readability / modifiability (and, maybe, quality) and use them in functions, where they improve quality / readability / modifiability.

For example, currently, early returns should not be used in functions, working with completion values.